### ISSUE TITLE
[A11y]: All headings should have semantic HTML

### ISSUE THREAD

**Issue by st0nebreaker** [sourcegraph/about#5544]
### Audit type

Screen reader navigation

### User journey audit issue

#5500

### Problem description

Many of these subtitle / tag-like elements are not using semantic HTML, so they are confusing when picked up by a screen reader. It's unclear to the user what order or significance the content is in.

![image](https://user-images.githubusercontent.com/59381432/176735283-4bcbd0cf-604b-466c-9a75-b5903ceffc71.png)
![image](https://user-images.githubusercontent.com/59381432/176735302-95a212a4-5bbe-40f3-810c-35bd43b82561.png)

### Expected behavior

Ensure all these subtitle / tag-like elements are using a header tag. If the tag is in the hero, the `h1` still needs to be the first item, so declare the `h1` first in the markup, and use flex-blox `flex-direction: column-reverse` to change the visual order.
